File: backend/business-module-persistence-commons/src/main/java/de/eshg/domain/model/BaseEntity.java
/*
 * Copyright 2024 cronn GmbH
 * SPDX-License-Identifier: Apache-2.0
 */

package de.eshg.domain.model;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;

@MappedSuperclass
public abstract class BaseEntity extends GenericEntity<Long> {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Override
  public Long getId() {
    return id;
  }

  public void setId(Long id) {
    this.id = id;
  }
}

File: employee-portal/src/lib/shared/api/models/BaseEntity.ts
/**
 * Copyright 2024 cronn GmbH
 * SPDX-License-Identifier: Apache-2.0
 */

export interface BaseEntity {
  readonly id: string;
}

interface BaseEntityProps {
  id: string;
}

export function mapBaseEntity(response: BaseEntityProps): BaseEntity {
  return { id: response.id };
}
